### ISSUE TITLE
Generate OptimInstance$search_space the same way TuningInstance does

### ISSUE THREAD

**Issue by mb706** [mlr-org/bbotk#135]
Otoh it would be nice to have consistency with `TuningInstance` here and use `objective$domain$search_space` as default for the search space for OptimInstance. OTOH it would break code that assumes the whole domain becomes the search space -- which is a reasonable assumption, tuning over the whole search space is probably the default in many cases.

We could check if `domain$search_space` is the empty `ParamSet`; if it is non-empty, we use the `domain$search_space`, if it is empty, then we use the whole `ParamSet`.

Alternatively we could initialize the `$domain` of every `OptimInstance` to have all `$values` set to `to_tune()`. However, that would break for `ParamUty`. If we then say that `ParamUty` just does not get initialized, then the default behaviour is to tune over all search space components except the `ParamUty`s, which is not nice.

So, how about: We initialize all `$domain$values` with `to_tune()`, except the `ParamUty`. Also, if the `domain` has a `ParamUty` given, then the `search_space` argument of `OptimInstance$initialize` *must* be given, otherwise it is an error. This is a reasonable thing to do in any case, since we usually can't tune over `ParamUty`, so having any default `search_space` here is probably broken to begin with.

**Comment by be-marc** [comment]
I like the first proposal. Do we want to merge https://github.com/mlr-org/bbotk/pull/138?

**Comment by mb706** [comment]
sure, although see my comment there about calling `search_space()` only once.

**Comment by be-marc** [comment]
Closed by #138
